cirq/circuits/circuit.py

```diff
@@ -1528,10 +1531,23 @@ def _get_operation_circuit_diagram_info_with_fallback(
 
     return protocols.CircuitDiagramInfo(wire_symbols=symbols)
 
+def _is_valid_identifier(text):
+    return re.match('[a-zA-Z_][a-zA-Z0-9_]*$', text)
+
+
+def _encode(text):
+    return json.JSONEncoder().encode(text)
 
 def _formatted_exponent(info: protocols.CircuitDiagramInfo,
                         args: protocols.CircuitDiagramInfoArgs
                         ) -> Optional[str]:
+
+    if isinstance(info.exponent, sympy.Basic):
+        name = str(info.exponent.free_symbols.pop())
```

Comment:
Why not just `name = str(info.exponent)`? E.g. this would give the right value of "a + b" when the exponent is `sympy.Symbol('a') + sympy.Symbol('b')`, instead of just "a".

Add a test that setting the exponent to `sympy.Symbol('a') + sympy.Symbol('b')` gives a reasonable looking circuit diagram.